test/server/http_test_server_filter_integration_test.cc

```diff
@@ -189,6 +189,24 @@ TEST_P(HttpTestServerIntegrationTest, TestHeaderConfig) {
   EXPECT_EQ(std::string(10, 'a'), response->body());
 }
 
+TEST_P(HttpTestServerIntegrationTest, TestEchoHeaders) {
```

Comment:
Can we add a test case with multiple requests and different headers in them to confirm the lifetime question above? The headers should be reflected just to a single response related to a single request.

Reply:
I've added unique_header to verify that echoed headers vary per request. 
Unfortunately I don't know how to make those requests parallel instead of sequential. Do you have any pointers?